types/mempool/priority_nonce.go

```diff
@@ -16,6 +15,40 @@ var (
 	_ Iterator = (*PriorityNonceIterator)(nil)
 )
 
+type TxPriority struct {
+	// GetTxPriority returns the priority of the transaction. A priority must be
+	// comparable via CompareTxPriority.
+	GetTxPriority func(ctx context.Context, tx sdk.Tx) any
+	// CompareTxPriority compares two transaction priorities. The result should be
+	// 0 if a == b, -1 if a < b, and +1 if a > b.
+	CompareTxPriority func(a, b any) int
+}
+
+// NewDefaultTxPriority returns a TxPriority comparator using ctx.Priority as
+// the defining transaction priority.
+func NewDefaultTxPriority() TxPriority {
+	return TxPriority{
+		GetTxPriority: func(goCtx context.Context, tx sdk.Tx) any {
+			return sdk.UnwrapSDKContext(goCtx).Priority()
+		},
+		CompareTxPriority: func(a, b any) int {
+			switch {
+			case a == nil && b == nil:
+				return 0
+			case a == nil:
```

Comment:
What's the logic for considering nil to be less than something?

Reply:
See https://github.com/cosmos/cosmos-sdk/pull/15328#discussion_r1134648037